packages/common/src/types/transactionEvent.ts

```diff
@@ -0,0 +1,352 @@
+        | TransactionEventTag.DelegationRemoved;
+    delegatorId: number;
+    account?: Base58String;
+}
+
+export interface DelegationSetDelegationTargetEvent {
+    tag: TransactionEventTag.DelegationSetDelegationTarget;
+    delegatorId: number;
+    account?: Base58String;
+    delegationTarget: EventDelegationTarget;
+}
+
+export interface DelegationSetRestakeEarningsEvent {
+    tag: TransactionEventTag.DelegationSetRestakeEarnings;
+    delegatorId: number;
+    account?: Base58String;
+    restakeEarnings: boolean;
+}
+
+export interface DelegationStakeChangedEvent {
+    tag:
+        | TransactionEventTag.DelegationStakeDecreased
+        | TransactionEventTag.DelegationStakeIncreased;
+    delegatorId: number;
+    account?: Base58String;
+    newStake: bigint;
+}
+
+// Baker Events
+
+export interface BakerAddedEvent {
+    tag: TransactionEventTag.BakerAdded;
+    bakerId: number;
+    account: string;
+    signKey: string;
+    electionKey: string;
+    aggregationKey: string;
+    stake: bigint;
+    restakeEarnings: boolean;
+}
+
+export interface BakerRemovedEvent {
+    tag: TransactionEventTag.BakerRemoved;
+    bakerId: number;
+    account?: Base58String;
+}
+
+export interface BakerStakeChangedEvent {
+    tag:
+        | TransactionEventTag.BakerStakeIncreased
+        | TransactionEventTag.BakerStakeDecreased;
+    bakerId: number;
+    account?: Base58String;
+    newStake: bigint;
+}
+
+export interface BakerSetRestakeEarningsEvent {
+    tag: TransactionEventTag.BakerSetRestakeEarnings;
+    bakerId: number;
+    account?: Base58String;
+    restakeEarnings: boolean;
+}
+
+export interface BakerKeysUpdatedEvent {
+    tag: TransactionEventTag.BakerKeysUpdated;
+    bakerId: number;
+    account: Base58String;
+    signKey: HexString;
+    electionKey: HexString;
+    aggregationKey: HexString;
+}
+
+export interface BakerSetOpenStatusEvent {
+    tag: TransactionEventTag.BakerSetOpenStatus;
+    bakerId: number;
+    account?: Base58String;
+    openStatus: OpenStatusText;
+}
+
+export interface BakerSetMetadataURLEvent {
+    tag: TransactionEventTag.BakerSetMetadataURL;
+    bakerId: number;
+    account?: Base58String;
+    metadataURL: string;
+}
+
+export interface BakerSetFinalizationRewardCommissionEvent {
+    tag: TransactionEventTag.BakerSetFinalizationRewardCommission;
+    bakerId: number;
+    account?: Base58String;
+    finalizationRewardCommission: number;
+}
+
+export interface BakerSetBakingRewardCommissionEvent {
+    tag: TransactionEventTag.BakerSetBakingRewardCommission;
+    bakerId: number;
+    account?: Base58String;
+    bakingRewardCommission: number;
+}
+
+export interface BakerSetTransactionFeeCommissionEvent {
+    tag: TransactionEventTag.BakerSetTransactionFeeCommission;
+    bakerId: number;
+    account?: Base58String;
+    transactionFeeCommission: number;
+}
+
+export interface UpdateEnqueuedEvent {
+    tag: TransactionEventTag.UpdateEnqueued;
+    effectiveTime: string;
```

Comment:
Don't you have a type for time strings?

Reply:
We have a digitString, but the v1 type actually just a number, I'll change it. :thinking: